scripts/samples/bench_and_plot.py

```diff
@@ -0,0 +1,354 @@
+# Copyright 2023 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Benchmark and plot sample access times across kinds of data and formats."""
+
+import os
+import string
+from argparse import ArgumentParser, Namespace
+from contextlib import contextmanager
+from shutil import rmtree
+from time import time, time_ns
+from typing import Any, Callable, Dict, Iterator, List, Tuple
+
+import numpy as np
+from matplotlib import pyplot as plt
+from matplotlib.ticker import ScalarFormatter
+from numpy.typing import DTypeLike, NDArray
+from tqdm import trange
+
+from streaming import CSVWriter, JSONWriter, MDSWriter, StreamingDataset
+
+
+def parse_args() -> Namespace:
+    """Parse command-line arguments.
+
+    Returns:
+        Namespace: Command-line arguments.
+    """
+    args = ArgumentParser()
+    args.add_argument('--data', type=str, default='data')
```

Comment:
Can you add the `help` args for each argument here? most of them are self explanatory except few.

Reply:
done